src/state/apis/swapper/selectors.ts

```diff
@@ -0,0 +1,14 @@
+import type { ReduxState } from 'state/reducer'
+
+export const selectSwapperApiPending = (state: ReduxState) =>
+  Object.values(state.swapperApi.queries).some(query => query?.status === 'pending')
+
+export const selectSwapperApiTradeQuotePending = (state: ReduxState) =>
+  Object.values(state.swapperApi.queries).some(
+    query => query?.endpointName === 'getTradeQuote' && query?.status === 'pending',
+  )
+
+export const selectSwapperApiUsdRatesPending = (state: ReduxState) =>
+  Object.values(state.swapperApi.queries).some(
+    query => query?.endpointName === 'getUsdRates' && query?.status === 'pending',
+  )
```

Comment:
The crux of this PR.

Reply:
LGTM, confirmed RTK doesn't have any notion of an aggregated global queries status of sort and this is the way to go
https://stackoverflow.com/questions/71523851/app-wide-rtk-query-loading-fetching-indicator